docs/docs/how_to/using-devcontainers.mdx

```diff
@@ -0,0 +1,108 @@
+---
+title: Devcontainers and Codespaces
+description: "Learn how to set up a devcontainer in your GitHub repository for a seamless coding experience with Codespaces. Follow our easy 8-step guide to create your own Noir environment without installing Nargo locally."
+keywords: ["Devcontainer", "Codespaces", "GitHub", "Noir Environment", "Docker Image", "Development Environment", "Remote Coding", "GitHub Codespaces", "Noir Programming", "Nargo", "VSCode Extensions", "Noirup"]
+sidebar_position: 1
+---
+
+Noir has extensive tooling. Thanks to Noirup, adding a devcontainer to your own repository is quite easy and pretty powerful.
+
+## What's a devcontainer after all?
+
+A devcontainer is a Docker image that comes preloaded with tools, extensions, and other tools you need to quickly get started or continue a project, without having to install Nargo locally. Think of it as a development environment in a box.
+
+There are many advantages to this:
+
+- It's platform and architecture agnostic
+- You don't need to have an IDE installed, or Nargo, or use a terminal at all
```

Comment:
Don't people still need to use a terminal, even if they are using a codespace?

Reply:
Yeah some things aren't yet on the vscode extension codelens, so you need to run nargo commands anyway... For now 😈

Reply:
```suggestion
- You don't need anything but a web browser to start coding
```